cli/lsp/documents.rs

```diff
@@ -1037,10 +1071,19 @@ impl DocumentsInner {
     specifier: &ModuleSpecifier,
     navigation_tree: Arc<tsc::NavigationTree>,
   ) -> Result<(), AnyError> {
```

Comment:
This `set_navigation_tree` method is somewhat prone to a race condition in which a navigation tree could be generated from a document on the file system, then updated on the file system, then this could set a navigation tree that's out of date. I think it would be extremely rare to happen if it ever did (because it would require an FS refresh), but I will prevent it from happening anyway in this PR soon.

Reply:
Done in 13ca3bd29b7b7e4d77247fc68b0509585cd71291